docs/ci-cd/README.md

```diff
@@ -0,0 +1,87 @@
+# CI/CD Pipeline with Taskcluster
+
+The code review bot relies **extensively** on Taskcluster, to run publication tasks, code analysis, but also to build and ship the code-review bot itself.
+
+We use both Mozilla's Taskcluster instances:
+
+- [firefox-ci](https://firefox-ci-tc.services.mozilla.com/) to build and run the bot project
+- [community](https://community-tc.services.mozilla.com/) to build the other projects
+
+We use a pretty simple convention related to the git workflow:
+
+1. Every pull request triggers a pipeline of tasks check unit tests, code format, ultimately building docker images.
+2. The `master` branch must always be *clean* and deployable. That branch is protected.
+3. Tags are only used to represent a release, using semantic versioning (example on [release 1.2.1](https://github.com/mozilla/code-review/releases/tag/1.2.1))
+4. A push on `testing` will run the full pipeline, building Docker images, and deploying them on the different providers for the **testing** environment (Taskcluster & Heroku)
+5. A push on `production` will run the full pipeline, building Docker images, and deploying them on the different providers for the **production** environment (Taskcluster & Heroku)
+
+
+## Firefox-CI instance
+
+![](firefox-ci.png)
+[Graph source code](firefox-ci.mermaid)
+
+The CI/CD workflow is really simple on the firefox-ci instance. On every Github push (branch, pull request, tag), we run the following tasks:
+
+- `check_lint` checks the formatting of all files
+- `bot_check_tests` runs the bot unit tests
+- `bot_build_dind` builds the bot's Docker image using Docker In Docker (dind), stores the image as a public artifact, and the task is always indexed.
+
+That last task is interesting because it allows us to ship immediately updates for the code review bot in the Taskcluster firefox-ci instance without relying on a remote Docker repository: the image is built in Taskcluster and stays there. Other tasks can use our doker image by referencing it through the **task-image** feature.
```

Comment:
```suggestion
That last task is interesting because it allows us to ship immediately updates for the code review bot in the Taskcluster firefox-ci instance without relying on a remote Docker repository: the image is built in Taskcluster and stays there. Other tasks can use our docker image by referencing it through the **task-image** feature.
```